3rd-party/class.jetpack-amp-support.php

```diff
@@ -0,0 +1,315 @@
+		add_filter( 'post_flair_disable',  array( 'Jetpack_AMP_Support', 'is_amp_canonical' ), 99 );
+	}
+
+	static function init_filter_jetpack_widgets() {
+		if ( ! self::is_amp_request() ) {
+			return;
+		}
+
+		// widgets
+		add_filter( 'jetpack_widgets_to_include', array( 'Jetpack_AMP_Support', 'filter_available_widgets' ) );
+	}
+
+	static function is_amp_canonical() {
+		return function_exists( 'amp_is_canonical' ) && amp_is_canonical();
+	}
+
+	static function is_amp_request() {
+		// can't use is_amp_endpoint() since it's not ready early enough in init.
+		// is_amp_endpoint() implementation calls is_feed, which bails with a notice if plugins_loaded isn't finished
+		// "Conditional query tags do not work before the query is run"
+		$is_amp_request = ! is_admin() // this is necessary so that modules can still be enabled/disabled/configured as per normal via Jetpack admin
+			&&
+				function_exists( 'amp_is_canonical' ) // this is really just testing if the plugin exists
+			&&
+				( amp_is_canonical() || isset( $_GET[ amp_get_slug() ] ) );
+
+		/**
+		 * Returns true if the current request should return valid AMP content.
+		 *
+		 * @since 6.2.0
+		 *
+		 * @param boolean $is_amp_request Is this request supposed to return valid AMP content?
+		 */
+		return apply_filters( 'jetpack_is_amp_request', $is_amp_request );
+	}
+
+	static function filter_available_widgets( $widgets ) {
+		if ( self::is_amp_request() ) {
+			$widgets = array_filter( $widgets, array( 'Jetpack_AMP_Support', 'is_supported_widget' ) );
+		}
+
+		return $widgets;
+	}
+
+	static function is_supported_widget( $widget_path ) {
+		return substr($widget_path, -14) !== '/milestone.php';
+	}
+
+	static function amp_disable_the_content_filters() {
+		if ( defined( 'WPCOM') && WPCOM ) {
+			add_filter( 'videopress_show_2015_player', '__return_true' );
+			add_filter( 'protected_embeds_use_form_post', '__return_false' );
+			remove_filter( 'the_title', 'widont' );
+		}
+
+		remove_filter( 'pre_kses', array( 'Filter_Embedded_HTML_Objects', 'filter' ), 11 );
+		remove_filter( 'pre_kses', array( 'Filter_Embedded_HTML_Objects', 'maybe_create_links' ), 100 );
+	}
+
+	/**
+	 * Add publisher and image metadata.
+	 *
+	 * @since 0.3
+	 *
+	 * @param array   $metadata Metadata array.
+	 * @param WP_Post $post     Post.
+	 * @return array Modified metadata array.
+	 */
+	static function amp_post_template_metadata( $metadata, $post ) {
+		if ( isset( $metadata['publisher'] ) && ! isset( $metadata['publisher']['logo'] ) ) {
+			$metadata = self::add_blavatar_to_metadata( $metadata );
+		}
+
+		if ( ! isset( $metadata['image'] ) ) {
+			$metadata = self::add_image_to_metadata( $metadata, $post );
+		}
+
+		return $metadata;
+	}
+
+	/**
+	 * Add blavatar to metadata.
+	 *
+	 * @since 0.3
+	 *
+	 * @param array $metadata Metadata.
+	 * @return array Metadata.
+	 */
+	static function add_blavatar_to_metadata( $metadata ) {
+		if ( ! function_exists( 'blavatar_domain' ) ) {
```

Comment:
Since blavatar is a .com thing. 
we could use the site_icon function here instead. 
Jetpack_Sync_Functions::site_icon_url()